Code/Framework/AzCore/AzCore/Jobs/Internal/JobTypeEraser.h

```diff
@@ -0,0 +1,232 @@
+            }
+            else
+            {
+                return reinterpret_cast<JobDestroy_t>(Destroyer);
+            }
+        }
+
+    private:
+        constexpr static void Invoker(Lambda* lambda)
+        {
+            lambda->operator()();
+        }
+
+        constexpr static void Mover(Lambda* dst, Lambda* src)
+        {
+            new (dst) Lambda{ AZStd::move(*src) };
+        }
+
+        constexpr static void Copyer(Lambda* dst, Lambda* src)
+        {
+            new (dst) Lambda{ *src };
+        }
+
+        constexpr static void Destroyer(Lambda* lambda)
+        {
+            lambda->~Lambda();
+        }
+    };
+
+    // The TypeErasedJob encapsulates member function pointers to store in a homogeneously-typed container
+    // The function signature of all lambdas encoded in a TypeErasedJob is void(*)(). The lambdas can capture
+    // data, in which case the data is inlined in this structure if the payload is less than or equal to the
+    // buffer size. Otherwise, the data is heap allocated.
+    class alignas(alignof(max_align_t)) TypeErasedJob final
+    {
+    public:
+        // The inline buffer allows the TypeErasedJob to span two cache lines. Lambdas can capture 48
+        // bytes of data (6 pointers/references on a 64-bit machine) before spilling to the heap.
+        constexpr static size_t BufferSize =
+            128 - sizeof(size_t) * 6 - sizeof(uint32_t) - sizeof(JobDescriptor) - sizeof(AZStd::atomic<uint32_t>);
+
+        TypeErasedJob() = default;
+
+        template<typename Lambda>
+        TypeErasedJob(JobDescriptor const& desc, Lambda&& lambda) noexcept
+            : m_descriptor{ desc }
+        {
+            JobTypeEraser<Lambda> eraser;
+            m_invoker = eraser.ErasedInvoker();
+            m_relocator = eraser.ErasedRelocator();
+            m_destroyer = eraser.ErasedDestroyer();
+
+            // NOTE: This code is conservative in that extended alignment requirements result in a heap
+            // spill, even if the lambda could have occupied a portion of the inline buffer with a base
+            // pointer adjustment.
+            if constexpr (sizeof(Lambda) <= BufferSize && alignof(Lambda) <= alignof(max_align_t))
+            {
+                TypedRelocate(AZStd::forward<Lambda>(lambda), m_buffer);
+                m_lambda = m_buffer;
+            }
+            else
+            {
+                // Lambda has spilled to the heap (or requires extended alignment)
+                m_lambda = reinterpret_cast<char*>(azmalloc(sizeof(Lambda), alignof(Lambda)));
+                TypedRelocate(AZStd::forward<Lambda>(lambda), m_lambda);
+            }
+        }
+
+        TypeErasedJob(TypeErasedJob&& other) noexcept;
+
+        TypeErasedJob& operator=(TypeErasedJob&& other) noexcept;
+
+        ~TypeErasedJob();
+
+        void Link(TypeErasedJob& other);
+
+        // Indicates if this job is a root of the graph (with no dependencies)
+        bool IsRoot();
```

Comment:
This function can be `const`

Reply:
Marking `const` but not `constexpr` since the inbound link count is almost certainly going to be changing at runtime dynamically